app/models/html_document.rb

```diff
@@ -87,26 +87,35 @@ def extract_metadata
     metadata = {}
     meta_nodes = html.xpath('//meta')
     meta_nodes.each do |node|
-      (metadata[node['name'].downcase] = node['content']) if node['name']
-      (metadata[node['property'].downcase] = node['content']) if node['property']
+      property = node['name'] || node['property']
```

Comment:
This is the biggest change, which impacts both the new tags logic, as well as existing metadata parsing for other fields.  At present, `extract_metadata` cannot handle parallel description.  So, something completely valid like:

```
<meta property="article:tag" content="NIAMS">
<meta property="article:tag" content="bone research">
```

Would only result in a tags value of "bone research" (dropping/overwriting "NIAMS").  The `extract_metadata` method has been rewritten to construct an array of for duplicated names or properties.

While the `&.first`s peppered throughout to convert these arrays back to strings aren't great, it at least makes more explicit what we've always been doing - taking only one value even if multiple values are provided.  And, specifically in the case of tags, returning arrays allows us to collect all values from all tag source fields, dedupe, and preserve all values - whether presented in one meta field or sprinkled across many.

Feedback/questions welcome.

Reply:
Thanks for explaining this! Just to make sure I'm understanding: the new behavior you've implemented breaks out the way we treat _keywords_, but leaves other types of metadata the same?

1. Extracts and stores _all_ keywords as a deduped array. 
2. Retains the old behavior (extracting only the first of potentially many values) for all other metadata.

Is that right?

Reply:
Yes, so the higher level `extract_metadata` method now always returns an array for _everything_, but the lower level, metadata-specific methods (where the `&.first`s have been added) all directly convert those arrays (mostly single-item arrays) back to a string.

On current main against the url `https://search.gov` the `extract_metadata` method would return:

```
{"handheldfriendly"=>"True",
 "mobileoptimized"=>"320",
 "viewport"=>"width=device-width, initial-scale=1.0",
 "msapplication-tilecolor"=>"#da532c",
 "msapplication-config"=>"https://search.gov/assets/img/favicons/browserconfig.xml",
 "theme-color"=>"#ffffff",
 "generator"=>"Jekyll v4.2.2",
 "og:title"=>"Search.gov Home",
 "og:locale"=>"en_US",
 "description"=>
  "Search.gov powers thousands of federal government search boxes. Use it to improve the public’s search experience on your agency’s site.",
 "og:description"=>
  "Search.gov powers thousands of federal government search boxes. Use it to improve the public’s search experience on your agency’s site.",
 "og:url"=>"https://search.gov/",
 "og:site_name"=>"Search.gov",
 "og:type"=>"website",
 "twitter:card"=>"summary",
 "twitter:title"=>"Search.gov Home"}
```

But, on this branch, we get:

```
{"handheldfriendly"=>["True"],
 "mobileoptimized"=>["320"],
 "viewport"=>["width=device-width, initial-scale=1.0"],
 "msapplication-tilecolor"=>["#da532c"],
 "msapplication-config"=>["https://search.gov/assets/img/favicons/browserconfig.xml"],
 "theme-color"=>["#ffffff"],
 "generator"=>["Jekyll v4.2.2"],
 "og:title"=>["Search.gov Home"],
 "og:locale"=>["en_US"],
 "description"=>
  ["Search.gov powers thousands of federal government search boxes. Use it to improve the public’s search experience on your agency’s site."],
 "og:description"=>
  ["Search.gov powers thousands of federal government search boxes. Use it to improve the public’s search experience on your agency’s site."],
 "og:url"=>["https://search.gov/"],
 "og:site_name"=>["Search.gov"],
 "og:type"=>["website"],
 "twitter:card"=>["summary"],
 "twitter:title"=>["Search.gov Home"]}
 ```
 
 But, since I've added in array handling to the specific metadata methods (e.g. `description` or `title`), both branches ultimately return a string. (So, to point 2 in your questions, yes, the old behavior is retained).
 
 Main:
 ```
 [5] pry(main)> doc.description
=> "Search.gov powers thousands of federal government search boxes. Use it to improve the public’s search experience on your agency’s site."
[6] pry(main)> doc.title
=> "Search.gov Home | Search.gov"
 ```
 
 This branch:
 ```
 [3] pry(main)> doc.description
=> "Search.gov powers thousands of federal government search boxes. Use it to improve the public’s search experience on your agency’s site."
[4] pry(main)> doc.title
=> "Search.gov Home | Search.gov"
 ```